zulipterminal/ui_tools/buttons.py

```diff
@@ -25,7 +25,6 @@ def __init__(self, controller: Any, caption: str,
             prefix = prefix_character[1]
         else:
             prefix = prefix_character
-        assert len(prefix) in (0, 1)
```

Comment:
What's the reasoning behind this particular assert? The buttons don't render properly if the prefix character is >1.

I need numeric (and incremental) `prefix_character` for in-line message links which could be >1 (e.g. 10).

Reply:
This defines how the current button works, ie. it is not guaranteed to work outside those bounds. If you wish it to be able to handle more, then we likely want to refactor the class to work with longer text.

That said, you're not using the extra feature which the prefix allows, which from memory is just a specific style. If you don't need that then you can skip the prefix and just put it all in the text?

Reply:
Thanks for the pointer. Given the issues with `caption`, `prefix_character` and that the existing methods weren't needed, I ended up implementing a new `urwid.Button` subclass to better tailor my needs for the message links.